### PR TITLE
default to quay based observability resources

### PR DESCRIPTION
**Pull request by pb82**
With the coming version of KFM, the observability configuration changes and the observability resources are now obtained from quay instead of Github. `OBSERVABILITY_CONFIG_ACCESS_TOKEN` is no longer used. Github based resources are still supported, but the access token has to be provided as part of a new config file.

Instead, I would suggest to just update the KAS Installer to the new quay location.

NOTE: this can only be merged, once https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1647 has landed. The KFM Git ref also needs to be updated one available.

### CODE REVIEW

**Comment by miguelsorianod** [comment]
> With the coming version of KFM, the observability configuration changes and the observability resources are now obtained from quay instead of Github. `OBSERVABILITY_CONFIG_ACCESS_TOKEN` is no longer used. Github based resources are still supported, but the access token has to be provided as part of a new config file.
> 
> Instead, I would suggest to just update the KAS Installer to the new quay location.
> 
> NOTE: this can only be merged, once [bf2fc6cc711aee1a0c2a/kas-fleet-manager#1647](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1647) has landed. The KFM Git ref also needs to be updated one available.

Would the credentials to be able to get the quay image need to be set somehow in kas installer?

**Comment by miguelsorianod** [comment]
> @miguelsorianod No, the KFM creates the image pull secret in the Observability namespace to pull that image.

How does the KFM get the content for the image pull secret?

**Comment by miguelsorianod** [comment]
> it's the same dockerconfig.json that also gets added to the FSO namespace, KFM gets it from the service secret in Vault. We have added permissions to the resources repo to the robot account used in there.

And KAS Installer gets it from the service secret in Vault?

**Comment by MikeEdgar** [comment]
> And KAS Installer gets it from the service secret in Vault?

The pull secret is created by the installer in `<KFM Namespace>/kas-fleet-manager-image-pull-secret` based on a user-provided username/password.

**Comment by miguelsorianod** [comment]
> > And KAS Installer gets it from the service secret in Vault?
> 
> The pull secret is created by the installer in `<KFM Namespace>/kas-fleet-manager-image-pull-secret` based on a user-provided username/password.

Then my original question still applies, would the credentials, which we now know are set by kas installer, be able to get the quay image from the quay repository that contains the observability config?
Do we need to update the documentation of kas installer to make clear that it is a requirement that those credentials must have access to a specific quay repository?

**Comment by MikeEdgar** [comment]
A few additional notes for when https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1647 is merged:
- update the KFM default commit https://github.com/bf2fc6cc711aee1a0c2a/kas-installer/blob/c7e14461180a0fa53f30ba9be993f6370a29c9f5/kas-installer-defaults.env#L140
- remove occurrences of `OBSERVABILITY_CONFIG_ACCESS_TOKEN` (several throughout the repo)

File: e2e-test-config.sh [hunk 1]
```diff
@@ -75,7 +75,6 @@ fi
 cat <<EOF
 {
   "LAUNCH_KEY": "${USER}",
-  "GITHUB_TOKEN": "${OBSERVABILITY_CONFIG_ACCESS_TOKEN}",
```
[the remark this answers is not in the document]
Comment (pb82) [reply e2e-test-config.sh:78]: If I understand this part of the installer code correctly, it was only printing the configuration summary. There are no other references to `GITHUB_TOKEN` inside the KAS Installer.

**Comment by miguelsorianod** [comment]
The changes look good to me. However, we will need to continue the remaining cleanup of kas-installer and the pipelines with @MikeEdgar , as we still see some remaining elements to be cleant.
Let's merge and tag already to fix the e2e pipelines failing after the KFM PR that had been merged.

**Pull request merged**